### ISSUE TITLE
Access to  OperatorHub through proxy is not working

### ISSUE THREAD

**Issue by yeplaa** [code-ready/crc#898]
### General information

  * OS: Linux centos 7.7
  * Hypervisor: KVM 
  * Did you run `crc setup` before starting it (Yes/No)? YEs

## CRC version
```bash
# Put the output of `crc version`
crc version: 1.2.0+c2e3c0f
OpenShift version: 4.2.8 (embedded in binary)

  
## CRC status
```bash
# Put the output of `crc status`
CRC VM:          Running
OpenShift:       Running (v4.2.8)
Disk Usage:      16.73GB of 32.2GB (Inside the CRC VM)
Cache Usage:     11.01GB
Cache Directory: /home/centos/.crc/cache


## CRC config
```bash
# Put the output of `crc config view`
```

## Host Operating System
```bash
# Put the output of `cat /etc/os-release` in case of Linux
# put the output of `sw_vers` in case of Mac
# Put the output of `systeminfo` in case of Windows
cat /etc/os-release
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"



### Steps to reproduce

I have used your https://github.com/code-ready/crc/wiki/Use-proxy-with-CRC doc mandatory to deploy crc front of a proxy. My proxy configuration is deployed as indicated into the doc.
The crc deployment is good.

### Expected
I want deploy operator by OperatorHub.

### Actual

I have currently a problem with the OperatorHub with the "No OperatorHub Items Found" message.
In the Operatorsource configuration the endpoint access is KO : 

oc get opsrc   -n openshift-marketplace
NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS        MESSAGE                                                                                                                                 AGE
certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=certified-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving   24d
community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=community-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving   24d
redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=redhat-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving      24d

Bellow the ClusterOperator status :  
oc get co
NAME                                       VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE
authentication                             4.2.8     True        False         False      47h
cloud-credential                           4.2.8     True        False         False      24d
cluster-autoscaler                         4.2.8     True        False         False      24d
console                                    4.2.8     True        False         False      7d18h
dns                                        4.2.8     True        False         False      32m
image-registry                             4.2.8     True        False         False      2d2h
ingress                                    4.2.8     True        False         False      24d
insights                                   4.2.8     True        False         False      24d
kube-apiserver                             4.2.8     True        False         False      24d
kube-controller-manager                    4.2.8     True        False         False      24d
kube-scheduler                             4.2.8     True        False         False      24d
machine-api                                4.2.8     True        False         False      24d
machine-config                             4.2.8     True        False         False      24d
marketplace                                4.2.8     True        False         False      91m
monitoring                                 4.2.8     False       True          True       24d
network                                    4.2.8     True        False         False      24d
node-tuning                                4.2.8     True        False         False      2d2h
openshift-apiserver                        4.2.8     True        False         False      2d2h
openshift-controller-manager               4.2.8     True        False         False      2d2h
openshift-samples                          4.2.8     True        False         False      24d
operator-lifecycle-manager                 4.2.8     True        False         False      24d
operator-lifecycle-manager-catalog         4.2.8     True        False         False      24d
operator-lifecycle-manager-packageserver   4.2.8     True        False         False      25h
service-ca                                 4.2.8     True        False         False      24d
service-catalog-apiserver                  4.2.8     True        False         False      24d
service-catalog-controller-manager         4.2.8     True        False         False      24d
storage                                    4.2.8     True        False         False      24d

Have you ever see this problem with crc  ?
Thanks

**Comment by praveenkumar** [comment]
@yeplaa Hi there, looks like we also need to add the proxy configuration to `marketplace-operator` because it is not added by default when using the proxy resource. Can you do following and let us know if that works for you? 

1. Get the proxy info from proxy resource
```
$ oc get proxy -oyaml
---  status:
    httpProxy: http://<my_proxy>
    httpsProxy: http://<my_proxy>
    noProxy: localhost,.cluster.local,.svc,.testing,10.128.0.0/14,127.0.0.1,172.30.0.0/16,192.168.126.0/24,api-int.crc.testing,api.crc.testing,etcd-0.crc.testing,localhost
---
```
2. Update those entry to marketplace-operator deployment
```
$ oc edit deployment.apps/marketplace-operator -n openshift-marketplace (just copy paste what you have on the proxy resource as value)
---
        - name: RELEASE_VERSION
          value: 4.2.8
        - name: HTTP_PROXY
          value: http://<your_proxy>
        - name: HTTPS_PROXY
          value: http://<your_proxy>
        - name: NO_PROXY
          value: localhost,.cluster.local,.svc,.testing,10.128.0.0/14,127.0.0.1,172.30.0.0/16,192.168.126.0/24,api-int.crc.testing,api.crc.testing,etcd-0.crc.testing,localhost
        image: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:95a42bd22ccea4b001732322ea06ba578768e2db9a09d839dc120b7138c0dc96
----
```

3. Once you update the deployment wait for all the pods become running again, it will terminate old ones and create new one.
```
$ oc get pods -n openshift-marketplace
```

4. Now verify it is able to fetch the contents.
```
$ oc get opsrc -n openshift-marketplace
NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS      MESSAGE                                       AGE
certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Succeeded   The object has been successfully reconciled   25d
community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Succeeded   The object has been successfully reconciled   25d
redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Succeeded   The object has been successfully reconciled   25d
```

**Comment by yeplaa** [comment]
Hi,

Thanks @praveenkumar for your reply. It works now.  
Is it this update will be permanent ? Is it deployment update will reload by Operator configuration? 

Loïc

**Comment by praveenkumar** [comment]
> Is it this update will be permanent ? 

@yeplaa It will be permanent as long as you using same `crc` instance once you delete and start again it will go away and you again need to make changes.

> Is it deployment update will reload by Operator configuration?
Yes, any change you do in the operator configuration it will also propagated to operand in this case so it just update the config for proxy in each operators in marketplace.

**Comment by nharik25** [comment]
I have the same issue resolving quay.io....from marketplace-operator pod.
My openshift cluster is not using PROXY....directly connected to internet. Please advice...

# oc get opsrc
NAME                  TYPE          ENDPOINT              REGISTRY              DISPLAYNAME           PUBLISHER   STATUS        MESSAGE                                                                                                                                 AGE
certified-operators   appregistry   https://quay.io/cnr   certified-operators   Certified Operators   Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=certified-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving   2d4h
community-operators   appregistry   https://quay.io/cnr   community-operators   Community Operators   Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=community-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving   2d4h
redhat-operators      appregistry   https://quay.io/cnr   redhat-operators      Red Hat Operators     Red Hat     Configuring   Get https://quay.io/cnr/api/v1/packages?namespace=redhat-operators: dial tcp: lookup quay.io on 172.30.0.10:53: server misbehaving      2d4h

**Comment by praveenkumar** [comment]
@yeplaa Can we close this issue?

**Comment by yeplaa** [comment]
Hi,

Yes thanks for your reply.
You can close this issue.

Have a good day !

**Comment by cfergeau** [comment]
This is probably also going to be an issue with the automatic proxy setup we added in our latest release, we should keep this open until this is fixed. Unless we mark it as a duplicate of #889.

**Comment by gbraad** [comment]
Likely closing this and adding a comment/todo on #889 would be appropriate

**Comment by cfergeau** [comment]
Fixed by https://github.com/code-ready/crc/commit/f8441be1f44043bb6a4367a9a5595f48430fbe2b
